### ISSUE TITLE
Tab completion is buggy with sdkman last versions

### ISSUE THREAD

**Issue by matthieusb** [matthieusb/zsh-sdkman#11]
Since sdkman last releases, the way `sdk list` is formatted and `sdk <program> list` changed.
This should be taken into account.

**Comment by matthieusb** [comment]
Update: this actually happends mostly with java. Todo:

- Check if it happens with other candidates

**Comment by manuelprinz** [comment]
Hi @matthieusb, although #11 works fine for me, I realized this was not the bug you addressed in this issue (listing all available versions). This should be fixed by using a different regex. The following should work on all OSes (but I tested only on Linux:)
```zsh
echo $( sdk list java | grep '|' | cut -d\| -f6 | grep -v Identifier )
```
The "problem" is that this is only saves the list of all versions. The caching code currently also tries to parse the whole output. And generates the other cache files from that.

My suggestion would be to get rid of the other methods because the candidate list, current version and installed versions can be directly obtained from the file system. This is very fast, so I think the caching for those can be removed. It is then sufficient to cache all versions on a candidate only, because the deltas can easily be computed on the fly. (At least theoretically.) :wink: 

I started some work that you can find in #22. It is not enough to fix this issue (#11) but I will improve on that. Just wanted to get your thoughts on the approach.

Best,
Manuel
